src/ucp/proto/proto_common.inl

```diff
@@ -362,6 +362,14 @@ ucp_proto_request_pack_rkey(ucp_request_t *req, ucp_md_map_t md_map,
                   md_map);
     }
 
+    if ((memh->flags & UCP_MEMH_FLAG_HAS_GVA) &&
+        (ucp_ep_config(req->send.ep)->key.err_mode ==
+         UCP_ERR_HANDLING_MODE_PEER)) {
```

Comment:
Maybe extend and re-use `ucp_ep_config_err_mode_check_mismatch` adding message parameter?